src/Pos/Ssc/GodTossing/Arbitrary.hs

```diff
@@ -39,12 +42,54 @@ import           Pos.Util.Arbitrary               (Nonrepeating (..), makeSmall,
 -- Core
 ----------------------------------------------------------------------------
 
+-- | Wrapper over 'Commitment'. Creates an invalid Commitment w.r.t. 'verifyCommitment'.
+newtype BadCommitment = BadComm
+    { getBadComm :: Commitment
+    } deriving (Show, Eq)
+
+instance Arbitrary BadCommitment where
+    arbitrary = BadComm <$> do
+      Commitment <$> arbitrary <*> arbitrary <*> (arbitrary `suchThat` (not . null))
+
+-- | Wrapper over 'Opening'. Creates an invalid Opening w.r.t. 'verifyOpening'.
```

Comment:
I don't understand this wrapper.
1. `Opening` itself can't be valid or invalid. Validity of `Opening` can be determined only when we know a `Commitment`.
2. As a consequence, `BadOpening`'s `Arbitrary` doesn't have any logic.

I think this wrapper is redundant.